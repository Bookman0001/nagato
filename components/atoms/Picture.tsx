import styled, { css } from 'styled-components'

interface Props {
  src: string
  alt: string
  isCircle?: boolean
  width?: number
  height?: number
}

export default function CirclePicture({
  src,
  alt,
  isCircle = false,
  width = 100,
  height = 100,
}: Props) {
  return (
    <Container>
      <Image
        width={width}
        height={height}
        src={src}
        alt={alt}
        isCircle={isCircle}
      />
    </Container>
  )
}

const Container = styled.div`
  border-radius: 100%;
`

const Image = styled.img<{ src: string; alt: string; isCircle: boolean }>`
  width: ${(props) => props.width}px;
  height: ${(props) => props.height}px;
  src: ${(props) => props.src};
  alt: ${(props) => props.alt};
  ${(props) =>
    props.isCircle &&
    css`
      border-radius: 100%;
    `}
`
