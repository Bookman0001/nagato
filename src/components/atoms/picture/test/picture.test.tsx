import { shallow } from 'enzyme'

import Picture from '../'

test('Picture', () => {
  const component = shallow(
    <Picture
      src="https://placehold.jp/150x150.png"
      alt="test"
      width={150}
      height={150}
    />
  )
  expect(component).toMatchSnapshot()
})
