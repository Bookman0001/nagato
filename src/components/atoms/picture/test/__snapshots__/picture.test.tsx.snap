// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Picture 1`] = `
<styled.div>
  <styled.img
    alt="test"
    height={150}
    isCircle={false}
    src="https://placehold.jp/150x150.png"
    width={150}
  />
</styled.div>
`;
